### ISSUE TITLE
Rebaseworkflow

### ISSUE THREAD

**Issue by Memnarch** [SRombauts/UE4GitPlugin#37]
My Problem:
I have Development and Featurebranch(remote and locally) and i am using git-lfs.
I made changes to my featurebranch and using normal Rebaseworkflow, i want to rebase it on my local copy of development. However this has the following problems:

1)The UI in editor does not support anything besides pulling/pushing from/to the remote
2)I need to use my GitExtensions to start the rebase, which works but 
2a) Lots of "can not unlink old file xy" (since my editor is open)
2b) any conflict which occurs is not detected in the Mergetool(i can't click Prev/Next Conflict)
3)if i keep the editor closed while rebasing and conflicts occur, openng the ditor will not show assets being part of the conflict so opening the Mergetool is not possible(i know, this is not your problem but that's what i struggle with)

**Comment by Memnarch** [comment]
btw how do i iniate a merge? if i have to iniate a merge from git-commandline or gitextensions, i face the same problems as above:
-can not unlink files (while ue4 editor is open)
- asset not visible(if editor was closed during merge)

**Comment by SRombauts** [comment]
Hi,

First of all, the Source Control in UE4 has been built around Perforce:
1. with file locking, so no conflict *ever*. For Subversion they enforced the same Locking mechanism.
2. without any "feature branches" so no merge nor any rebase

To have a good experience with Git, you should try to use a workflow that avoid relying on branch/merge, and try to avoir conflicts by splitting the workload and using other collaborative tools.

I didn't try merge-conflicts in a while, but you have(had) to close the Editor, use you GUI/CLI git tool, than reopen the Editor to manage conflicted files. **Then you can only choose one version or the other to keep** , so this it not a real merge at all!
Let me know if this is not working as intended.

Cheers!

**Comment by Memnarch** [comment]
Okay if it is just pick a or b that it is already the same as with git itself -.-
If you reopen the editor during conflicts, you can not pick the assets anywhere.
Thanks for the answer

**Comment by SRombauts** [comment]
No, I mean, there IS a "Blueprint Visual Diff merge viewer" (and texte comparisons for other assets like Materials).
**But** the Visual Diff viewer is just a viewer showing what has been change in the Blueprint, help choose the "best" version,

**Comment by Memnarch** [comment]
I only know of the diff/merge tools visible when a blueprint is open but they always work only for the currently open blueprint. I know of no other option :/

**Comment by SRombauts** [comment]
When you create two git branches, modify the same blueprint on the two branches, and then merge those two branches together (with UE4 Editor closed) you get a conflict.

1. "git status" should says that the Content/Xxx/Yyy_BP.uasset file is in conflict.
2. then, launching the Editor, you should get a "Warning" icon on your Blueprint in the Content Browser
3. then the context menu should give you an option to "Solve" (or "Merge", I don't recall precisly) the conflict
4. this should open a window with a 3-way Merge Window (3 views) of your Blueprint, with colors showing the diffferences

Let me know if this is not working for you, and in which version of the Engine.

**Comment by SRombauts** [comment]
Also, it could be a problem with UE vs Git LFS! **I never tried to merge with Git LFS!**

**Comment by Memnarch** [comment]
Nope, Blueprint will not showup in ContentBrowser

**Comment by Memnarch** [comment]
AH ok, so with upcoming 4.16 it should work? (at least getting the conflicts in editor?)

**Comment by SRombauts** [comment]
I am not sure if will work since I never tried to merge with Git LFS.
The newest version of the plugin fixes the "visual diff" in the History of a Blueprint 

Also, you can try to install the plugin from my release page to replace the one shipped with Unreal Engine so that you can already try to see if it works for you. 

I will try and will update the plugin if needed!

**Comment by SRombauts** [comment]
Please, test the 2.0 release of the plugin, installing (clone or unzip) it in your project "Plugins" folder (create this folder if this is your first project plugin) 

Then report here if it works any better

**Comment by Memnarch** [comment]
Merges so far  don't show up so far, however i simply readded my changes afterwards(thus overriding those introduces from rebase) and tried to diff/merge. Was not able to load the bp for diffing :/

**Comment by SRombauts** [comment]
Could you send me your log please?

**Comment by SRombauts** [comment]
Git LFS visual diff in history (compare with previous commit) should work under Windows but would require Git 2.10 (README.md could be more precise about that)

**Comment by Memnarch** [comment]
I use 
git 2.12.0.windows.1
GitLFS 2.0.2
I'll look into sending logs later. Right now i started duplicating my assets to get the diffs

**Comment by danielfinke** [comment]
I think that I am having the same problem as @Memnarch. If a blueprint asset is marked as a conflict by Git, it won't be visible in the Content Browser at all. There is no way to get to the BP merge tool.
Furthermore, trying to use the tool on any non-conflicted BP gives the error:

> Aborted Load of "\<asset path\>" because we could not get the requested revision
> Failed to load asset(s) for merge.

git version 2.14.2.windows.1
git-lfs/2.3.0 (GitHub; windows amd64; go 1.8.3; git 4dd2bf73)
